e2e/dractions/disapp.go

```diff
@@ -25,10 +28,27 @@ func EnableProtectionDiscoveredApps(ctx types.TestContext) error {
 	placementName := name
 	drpcName := name
 
-	// We assume that the application is on the first cluster (c1).
-	// This is correct only for the current test flow.
-	// TODO: must discover where the application is located instead.
-	cluster := ctx.Env().C1
+	var cluster types.Cluster
+
+	statuses, err := w.Status(ctx)
+	if err != nil {
+		return err
+	}
+
+	switch len(statuses) {
+	case 0:
+		return fmt.Errorf("application \"%s/%s\" not found", appNamespace, appname)
+	case 1:
+		log.Debugf("Application \"%s/%s\" found in cluster %q with status %q",
+			appNamespace, appname, statuses[0].ClusterName, statuses[0].Status)
+
+		cluster, err = ctx.Env().GetCluster(statuses[0].ClusterName)
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("application \"%s/%s\" found on multiple clusters: %+v", appNamespace, appname, statuses)
+	}
```

Comment:
Same as previous change, instead of adding nolint, add this code in a helper that makes it clear what is the purpose of the check.